From 73cc2dd3cc873d485bbbf207f99eb4df5a8d71b9 Mon Sep 17 00:00:00 2001
From: Adam YH Lee <adam.yh.lee@gmail.com>
Date: Mon, 8 Dec 2014 13:27:21 -0800
Subject: [PATCH 32/32] omap3: overo: Add device tree for Arbor43C board

Add a device tree supporting the Gumstix Arbor43C expansion board [1]
for Gumstix Overo and Overo Storm COMs.  This board features a 4.3"
capacitive touchscreen but is otherwise rather similar to the now-
discontinued Chestnut43 board.

[1] https://store.gumstix.com/expansion/overo-expansion/arbor-43c.html

Signed-off-by: Ash Charles <ashcharles@gmail.com>
---
 arch/arm/boot/dts/Makefile                         |  2 +
 arch/arm/boot/dts/omap3-overo-arbor43c-common.dtsi | 69 ++++++++++++++++++++++
 arch/arm/boot/dts/omap3-overo-arbor43c.dts         | 38 ++++++++++++
 arch/arm/boot/dts/omap3-overo-common-lcd43.dtsi    |  1 +
 arch/arm/boot/dts/omap3-overo-common-lcd43c.dtsi   | 18 ++++++
 .../boot/dts/omap3-overo-common-peripherals.dtsi   | 20 +++++++
 arch/arm/boot/dts/omap3-overo-storm-arbor43c.dts   | 39 ++++++++++++
 arch/arm/mach-omap2/pdata-quirks.c                 |  1 -
 8 files changed, 187 insertions(+), 1 deletion(-)
 create mode 100644 arch/arm/boot/dts/omap3-overo-arbor43c-common.dtsi
 create mode 100644 arch/arm/boot/dts/omap3-overo-arbor43c.dts
 create mode 100644 arch/arm/boot/dts/omap3-overo-common-lcd43c.dtsi
 create mode 100644 arch/arm/boot/dts/omap3-overo-storm-arbor43c.dts

diff --git a/arch/arm/boot/dts/Makefile b/arch/arm/boot/dts/Makefile
index 2578485d6f6b..67fbda4548e4 100644
--- a/arch/arm/boot/dts/Makefile
+++ b/arch/arm/boot/dts/Makefile
@@ -309,11 +309,13 @@ dtb-$(CONFIG_ARCH_OMAP3) += am3517-craneboard.dtb \
 	omap3-n9.dtb \
 	omap3-n950.dtb \
 	omap3-overo-alto35.dtb \
+	omap3-overo-arbor43c.dtb \
 	omap3-overo-chestnut43.dtb \
 	omap3-overo-gallop43.dtb \
 	omap3-overo-palo35.dtb \
 	omap3-overo-palo43.dtb \
 	omap3-overo-storm-alto35.dtb \
+	omap3-overo-storm-arbor43c.dtb \
 	omap3-overo-storm-chestnut43.dtb \
 	omap3-overo-storm-gallop43.dtb \
 	omap3-overo-storm-palo35.dtb \
diff --git a/arch/arm/boot/dts/omap3-overo-arbor43c-common.dtsi b/arch/arm/boot/dts/omap3-overo-arbor43c-common.dtsi
new file mode 100644
index 000000000000..d869480924bd
--- /dev/null
+++ b/arch/arm/boot/dts/omap3-overo-arbor43c-common.dtsi
@@ -0,0 +1,69 @@
+/*
+ * Copyright (C) 2014 Gumstix, Inc. - https://www.gumstix.com/
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 as
+ * published by the Free Software Foundation.
+ */
+
+/*
+ * Arbor43C expansion board is manufactured by Gumstix Inc.
+ * https://store.gumstix.com/index.php/products/635/
+ */
+
+#include "omap3-overo-common-peripherals.dtsi"
+#include "omap3-overo-common-lcd43c.dtsi"
+
+#include <dt-bindings/input/input.h>
+
+/ {
+	leds {
+		compatible = "gpio-leds";
+		pinctrl-names = "default";
+		pinctrl-0 = <&led_pins>;
+		heartbeat {
+			label = "overo:red:gpio163";
+			gpios = <&gpio6 3 GPIO_ACTIVE_HIGH>;		/* gpio_163 */
+			/* linux,default-trigger = "heartbeat"; */
+		};
+		gpio150 {
+			label = "overo:blue:gpio150";
+			gpios = <&gpio5 22 GPIO_ACTIVE_HIGH>;		/* gpio_150 */
+		};
+		gpio186 {
+			label = "overo:yellow:gpio186";
+			gpios = <&gpio6 26 GPIO_ACTIVE_HIGH>; 	/* gpio_186 */
+		};
+	};
+
+	gpio_keys {
+		compatible = "gpio-keys";
+		pinctrl-names = "default";
+		pinctrl-0 = <&button_pins>;
+		#address-cells = <1>;
+		#size-cells = <0>;
+		button1@114 {
+			label = "button1";
+			linux,code = <BTN_1>;
+			gpios = <&gpio4 18 GPIO_ACTIVE_LOW>;		/* gpio_114 */
+			gpio-key,wakeup;
+		};
+	};
+};
+
+#include "omap-gpmc-smsc9221.dtsi"
+
+&gpmc {
+	ranges = <5 0 0x2c000000 0x1000000>;	/* CS5 */
+
+	ethernet@gpmc {
+		reg = <5 0 0xff>;
+		interrupt-parent = <&gpio6>;
+		interrupts = <16 IRQ_TYPE_LEVEL_LOW>;	/* GPIO 176 */
+	};
+};
+
+&lis33de {
+	status = "disabled";
+};
+
diff --git a/arch/arm/boot/dts/omap3-overo-arbor43c.dts b/arch/arm/boot/dts/omap3-overo-arbor43c.dts
new file mode 100644
index 000000000000..203e24a5bbc3
--- /dev/null
+++ b/arch/arm/boot/dts/omap3-overo-arbor43c.dts
@@ -0,0 +1,38 @@
+/*
+ * Copyright (C) 2014 Gumstix, Inc. - https://www.gumstix.com/
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 as
+ * published by the Free Software Foundation.
+ */
+
+/*
+ * Arbor43C expansion board is manufactured by Gumstix Inc.
+ * https://store.gumstix.com/index.php/products/635/
+ */
+
+/dts-v1/;
+
+#include "omap3-overo.dtsi"
+#include "omap3-overo-arbor43c-common.dtsi"
+
+/ {
+	model = "OMAP35xx Gumstix Overo on Arbor43C";
+	compatible = "gumstix,omap3-overo-arbor43c", "gumstix,omap3-overo", "ti,omap3430", "ti,omap3";
+};
+
+&omap3_pmx_core2 {
+	led_pins: pinmux_led_pins {
+		pinctrl-single,pins = <
+			OMAP3630_CORE2_IOPAD(0x21E0, PIN_OUTPUT | MUX_MODE4)	/* sys_nirq.gpio_186 */
+			OMAP3630_CORE2_IOPAD(0x2198, PIN_OUTPUT | MUX_MODE4)	/* mcbsp1_clkx.gpio_168 */
+			OMAP3630_CORE2_IOPAD(0x2180, PIN_OUTPUT | MUX_MODE4)	/* uart1_cts.gpio_150 */
+		>;
+	};
+
+	button_pins: pinmux_button_pins {
+		pinctrl-single,pins = <
+			OMAP3630_CORE2_IOPAD(0x25dc, PIN_INPUT | MUX_MODE4)		/* csi2_dx1.gpio114 */
+		>;
+	};
+};
diff --git a/arch/arm/boot/dts/omap3-overo-common-lcd43.dtsi b/arch/arm/boot/dts/omap3-overo-common-lcd43.dtsi
index 048fd216970a..606dfe83dca8 100644
--- a/arch/arm/boot/dts/omap3-overo-common-lcd43.dtsi
+++ b/arch/arm/boot/dts/omap3-overo-common-lcd43.dtsi
@@ -133,6 +133,7 @@
 		regulator-name = "ads7846-reg";
 		regulator-min-microvolt = <3300000>;
 		regulator-max-microvolt = <3300000>;
+		status = "okay";
 	};
 
 	backlight {
diff --git a/arch/arm/boot/dts/omap3-overo-common-lcd43c.dtsi b/arch/arm/boot/dts/omap3-overo-common-lcd43c.dtsi
new file mode 100644
index 000000000000..89d7dbb1840c
--- /dev/null
+++ b/arch/arm/boot/dts/omap3-overo-common-lcd43c.dtsi
@@ -0,0 +1,18 @@
+/*
+ * Copyright (C) 2014 Gumstix, Inc. - https://www.gumstix.com/
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 as
+ * published by the Free Software Foundation.
+ */
+
+/*
+ * 4.3'' LCD Capacitive panel output for some Gumstix Overo boards (Arbor43C)
+ * https://store.gumstix.com/index.php/products/635/
+ */
+
+#include "omap3-overo-common-lcd43.dtsi"
+
+&ads7846reg {
+	status = "disabled";
+};
diff --git a/arch/arm/boot/dts/omap3-overo-common-peripherals.dtsi b/arch/arm/boot/dts/omap3-overo-common-peripherals.dtsi
index 7cfb808e8f5a..1e484b881b5b 100644
--- a/arch/arm/boot/dts/omap3-overo-common-peripherals.dtsi
+++ b/arch/arm/boot/dts/omap3-overo-common-peripherals.dtsi
@@ -40,6 +40,13 @@
 			OMAP3_CORE1_IOPAD(0x21a0, PIN_OUTPUT | MUX_MODE0)	/* uart3_tx_irtx.uart3_tx_irtx */
 		>;
 	};
+
+	ft5306_pins: pinmux_ft5306_pins {
+		pinctrl-single,pins = <
+			OMAP3_WKUP_IOPAD(0x2a18, PIN_INPUT | MUX_MODE4)         /* sys_clkout1.gpio_10 */
+			OMAP3_WKUP_IOPAD(0x2a26, PIN_OUTPUT | MUX_MODE4)        /* jtag_emu1.gpio_31 */
+		>;
+	};
 };
 
 &i2c3 {
@@ -82,6 +89,19 @@
 		st,max-limit-z = <750>;
 	};
 
+	captouch: edt-ft5306@38 {
+		compatible = "edt,edt-ft5306", "edt,edt-ft5x06";
+		pinctrl-names = "default";
+		pinctrl-0 = <&ft5306_pins>;
+
+		reg = <0x38>;
+		interrupt-parent = <&gpio1>;
+		interrupts = <10>;
+		wake-gpios = <&gpio1 31 GPIO_ACTIVE_LOW>;
+		touchscreen-size-x = <480>;
+		touchscreen-size-y = <272>;
+	};
+
 	/* optional Caspa (MT9V032) camera */
 	mt9v032_1: camera@5c {
 		compatible = "aptina,mt9v032";
diff --git a/arch/arm/boot/dts/omap3-overo-storm-arbor43c.dts b/arch/arm/boot/dts/omap3-overo-storm-arbor43c.dts
new file mode 100644
index 000000000000..6c0349189521
--- /dev/null
+++ b/arch/arm/boot/dts/omap3-overo-storm-arbor43c.dts
@@ -0,0 +1,39 @@
+/*
+ * Copyright (C) 2014 Gumstix, Inc. - https://www.gumstix.com/
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 as
+ * published by the Free Software Foundation.
+ */
+
+/*
+ * Arbor43C expansion board is manufactured by Gumstix Inc.
+ * https://store.gumstix.com/index.php/products/635/
+ */
+
+/dts-v1/;
+
+#include "omap3-overo-storm.dtsi"
+#include "omap3-overo-arbor43c-common.dtsi"
+
+/ {
+	model = "OMAP36xx/AM37xx/DM37xx Gumstix Overo on Arbor43C";
+	compatible = "gumstix,omap3-overo-arbor43c", "gumstix,omap3-overo", "ti,omap36xx", "ti,omap3";
+};
+
+&omap3_pmx_core2 {
+	led_pins: pinmux_led_pins {
+		pinctrl-single,pins = <
+			OMAP3630_CORE2_IOPAD(0x21E0, PIN_OUTPUT | MUX_MODE4)	/* sys_nirq.gpio_186 */
+			OMAP3630_CORE2_IOPAD(0x2198, PIN_OUTPUT | MUX_MODE4)	/* mcbsp1_clkx.gpio_168 */
+			OMAP3630_CORE2_IOPAD(0x2180, PIN_OUTPUT | MUX_MODE4)	/* uart1_cts.gpio_150 */
+		>;
+	};
+
+	button_pins: pinmux_button_pins {
+		pinctrl-single,pins = <
+			OMAP3630_CORE2_IOPAD(0x25dc, PIN_INPUT | MUX_MODE4)		/* csi2_dx1.gpio114 */
+		>;
+	};
+};
+
diff --git a/arch/arm/mach-omap2/pdata-quirks.c b/arch/arm/mach-omap2/pdata-quirks.c
index eb0c54da3363..8a499019a080 100644
--- a/arch/arm/mach-omap2/pdata-quirks.c
+++ b/arch/arm/mach-omap2/pdata-quirks.c
@@ -421,7 +421,6 @@ static struct pdata_init pdata_quirks[] __initdata = {
 	{ "ti,am3517-evm", am3517_evm_legacy_init, },
 	{ "technexion,omap3-tao3530", omap3_tao3530_legacy_init, },
 	{ "gumstix,omap3-overo", overo_legacy_init, },
-	{ "gumstix,omap3-overo-arbor43c", gumstix_arbor43c_legacy_init, },
 #endif
 #ifdef CONFIG_ARCH_OMAP4
 	{ "ti,omap4-sdp", omap4_sdp_legacy_init, },
-- 
2.1.4

